graphql/src/main/kotlin/com/trib3/graphql/resources/GraphQLResource.kt

```diff
@@ -161,13 +163,17 @@ open class GraphQLResource
     open fun graphQLUpgrade(
         @Parameter(hidden = true) @Auth principal: Optional<Principal>,
         @Context request: HttpServletRequest,
-        @Context response: HttpServletResponse
+        @Context response: HttpServletResponse,
+        @Context containerRequestContext: ContainerRequestContext
     ): Response {
-        if (graphQLConfig.checkAuthorization && !principal.isPresent) {
+        if (
+            request.getHeader("Origin") != null &&
+            response.getHeader(CrossOriginFilter.ACCESS_CONTROL_ALLOW_ORIGIN_HEADER) == null
+        ) {
```

Comment:
Does the actual comparing to allowed list happen in the checkAuth call?

Reply:
the actual comparing happened in the `CrossOriginFilter` jetty filter.  however, the next PR actually changes this because that filter doesn't set the header on websockets no matter what because of chrome: https://bugs.eclipse.org/bugs/show_bug.cgi?id=315190